### PR TITLE
Refactor/sfox bank link

### PR DESCRIPTION
**Pull request by prwelber**
## Description
I was looking into an issue reported by support but could not reproduce. Did some refactoring while I was in there. 

- removed redux-ui
- moved data gathering from mapStateToProps --> selectors.js
- use class properties and arrow functions so we can remove the constructor

## Change Type
Please enter one or more of the following: 
- Refactor

## Testing Steps
We'll need the wallet on testnet for QA to look at this. 
If that's ready, then to test this link a bank account to your sfox account.

## Code Checklist
- [ ] Code compiles successfully (verified via `yarn start`)
- [ ] No lint issues exist (verified via `yarn lint`)
- [ ] New and existing unit tests pass (verified via `yarn test`)
- [ ] `README.md` and other documentation is updated as needed

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/selectors.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    'web',
+    'sfox',
+    'config'
+  ]
+  const plaidPath = append('plaid', basePath)
+  const plaidEnvPath = append('plaidEnv', basePath)
+
+  const bankAccounts = selectors.core.data.sfox.getBankAccounts(state)
+  const accounts = selectors.core.data.sfox.getAccounts(state)
+  const deposit1 = formValueSelector('sfoxLink')(state, 'deposit1')
```
Comment (tony-blockchain) [inline packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/selectors.js]: formValueSelector could be run once at the outmost scope

Comment (tony-blockchain) [reply packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/selectors.js]: the first call that is*

File: packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/index.js [hunk 2]
```diff
@@ -143,49 +132,22 @@ class LinkContainer extends Component {
...
+        microStep={this.state.microStep}
+        goToMicroDepositStep={step => this.setState({ microStep: step })}
         submitMicroDeposits={this.submitMicroDeposits}
         awaitingDeposits={awaitingDeposits}
         showModal={showModal}
         busy={sfoxBusy}
         setNotAsked={sfoxNotAsked}
         linkError={err && path(['message'], err)}
         resetAccountHolder={this.resetAccountHolder}
+        componentState={this.state}
```
[the remark this answers is not in the document]
Comment (prwelber) [reply packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/index.js]: I don't think that's a confusing name, but I agree destructuring is a much better way to do this.

File: packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/selectors.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { selectors } from 'data'
+import { formValueSelector } from 'redux-form'
+import { path, append } from 'ramda'
+
+export const getData = state => {
+  const basePath = [
```
[the remark this answers is not in the document]
Comment (prwelber) [reply packages/blockchain-wallet-v4-frontend/src/modals/Sfox/SfoxExchangeData/Link/selectors.js]: created new selectors to replace paths.

**Comment by plondon** [comment]
Would it be possible to add some test coverage for any of the sfox sagas?

**Pull request merged**